### PR TITLE
Add logging via Katip

### LINKED ISSUE #17: Add logging via Katip
Haskell has a number of good libraries for logging. `katip` is a particularly fully featured one that is a great choice for production applications.

- [ ] Add the library to the dependencies
- [ ] Add the necessary fields to the `Config` data type.
- [ ] Write an instance for Katip's log class so that we can use logging in the `App` monad.

### PR DESCRIPTION
**Pull request by 0rca**
This hopefully resolves #17

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[*.hs]
+indent_style = space
+indent_size = 4
```
Comment (parsonsmatt) [inline .editorconfig]: awesome! thanks!

File: app/Main.hs [hunk 2]
```diff
@@ -1,21 +1,21 @@
 {-# LANGUAGE OverloadedStrings #-}
 module Main where
-
+import qualified Control.Monad.Metrics       as M
```
Comment (parsonsmatt) [inline app/Main.hs]: Style: leave a newline between `module Main where` and the import list.

File: src/Config.hs [hunk 3]
```diff
@@ -41,13 +43,26 @@ type App = AppT IO
 -- running in and a Persistent 'ConnectionPool'.
 data Config
     = Config
-    { getPool :: ConnectionPool
-    , getEnv  :: Environment
-    , getMetrics :: Metrics
+    { getPool      :: ConnectionPool
+    , getEnv       :: Environment
+    , getMetrics   :: Metrics
+    , configLogEnv :: LogEnv
```
Comment (parsonsmatt) [inline src/Config.hs]: I do prefer the new naming convention for the config's fields. However, for consistency's sake, if we're going to change them, then we should change all of them at once.

File: src/Config.hs [hunk 4]
```diff
@@ -58,21 +73,31 @@ data Environment
...
+setLogger Development env = katipLogger env
+setLogger Production env  = katipLogger env
+
+-- | Web request logger (currently unimplemented). For inspiration see
+-- ApacheLogger from wai-logger package.
+katipLogger :: LogEnv -> Middleware
+katipLogger env app req respond = runKatipT env $ do
+    -- todo: log proper request data
+    logMsg "web" InfoS "todo: received some request"
+    liftIO $ app req respond
```
Comment (parsonsmatt) [inline src/Config.hs]: I'm inclined to say that it should be fine to continue using the original request middleware loggers. The request middleware logging is distinct from the more structured app logging that we might want to do.

Comment (parsonsmatt) [reply src/Config.hs]: I would be happy to see a further PR that implemented the request middleware logging in katip; though that may even be worthy of it's own library :smile:

File: src/Config.hs [hunk 5]
```diff
@@ -58,21 +73,31 @@ data Environment
...
 -- insecure connection string. The 'Production' environment acquires the
 -- information from environment variables that are set by the keter
 -- deployment application.
 makePool :: Environment -> IO ConnectionPool
-makePool Test =
-    runNoLoggingT (createPostgresqlPool (connStr "-test") (envPool Test))
-makePool Development =
-    runStdoutLoggingT (createPostgresqlPool (connStr "") (envPool Development))
+makePool Test = do
+    env <- mkLogEnv
```
Comment (parsonsmatt) [inline src/Config.hs]: Rather than making a throwaway log pool, it seems like passing the parameter in would be cleaner.

File: src/Logger.hs [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+module Logger (
+    mkLogEnv,
+    runKatipT,
+    KatipT(..),
+    Katip(..),
+    LogEnv,
+    logMsg,
+    Severity(..),
+    adapt
+) where
```
Comment (parsonsmatt) [inline src/Logger.hs]: Style nit: prefer comma before the item in an import list.

```haskell
module Logger
    ( mkLogEnv
    , runKatipT
    , -- etc
    ) where
```

**Comment by 0rca** [comment]
Thanks for the nitpicks!

**Comment by parsonsmatt** [comment]
Just need to resolve the merge conflicts and this is good to merge. Thanks!

**Comment by 0rca** [comment]
I've done that, too. Enjoy!

**Pull request merged**